desc/objectives/_free_boundary.py

```diff
@@ -754,6 +755,338 @@ def _print(fmt, fmax, fmin, fmean, norm, units):
+        Collocation grid containing the nodes for plasma source terms.
+    eval_grid : Grid, optional
+        Collocation grid containing the nodes on the plasma surface at which the
+        magnetic field is being calculated and where to evaluate Bn errors.
+    field_grid : Grid, optional
+        Grid used to discretize ext_field (e.g. grid for the magnetic field source from
+        coils).
+    eq_fixed : bool
+        Whether or not the equilibrium's DOF (params) change during optimization.
+        If true, Bplasma will be precomputed in build().
+    field_fixed : bool
+        Whether or not the external field's DOF (params) change during optimization.
+    vacuum : bool
+        If true, Bplasma is set to zero.
+    name : str
+        Name of the objective function.
+
+    """
+
+    _scalar = False
+    _linear = False
+    _print_value_fmt = "Boundary normal field Error: {:10.3e} "
+    _units = "(T)"
+    _coordinates = "rtz"
+
+    def __init__(
+        self,
+        ext_field,
+        eq=None,
+        target=0,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        s=None,
+        q=None,
+        src_grid=None,
+        eval_grid=None,
+        field_grid=None,
+        eq_fixed=False,
+        field_fixed=False,
+        vacuum=False,
+        name="Quadratic flux",
+    ):
+        self._src_grid = src_grid
+        self._eval_grid = eval_grid
+        self._eq = eq
+        self._s = s
+        self._q = q
+        self._ext_field = ext_field
+        self._field_grid = field_grid
+        self._eq_fixed = eq_fixed
+        self._field_fixed = field_fixed
+        self._vacuum = vacuum
+        if not eq_fixed and not field_fixed:
+            things = [ext_field, eq]
+        elif eq_fixed and not field_fixed:
+            things = [ext_field]
+        elif field_fixed and not eq_fixed:
+            things = [eq]
+        else:
+            raise ValueError("Cannot fix both the eq and field.")
+        super().__init__(
+            things=things,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, eq=None, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        if self._eq_fixed:
+            eq = self._eq
+        elif self._field_fixed:
+            eq = self.things[0]
+        else:
+            eq = self.things[1]
+
+        if eq != self._eq:
+            self._eq = eq
+
+        if self._src_grid is None:
+            src_grid = LinearGrid(
```

Comment:
if the equilibrium is always fixed we can use a higher resolution grid here to ensure the plasma field is computed accurately.

Reply:
I set it to N=M=15

Reply:
It should still be based on `eq.M_grid` etc, but we can do `3*eq.M_grid` or something